libminifi/include/core/state/Value.h

```diff
@@ -46,26 +56,99 @@ class Value {
     return string_value;
   }
 
-  bool empty(){
+  template<typename T>
+  bool convertValue(T &ref) {
+    return convertValueImpl<typename std::common_type<T>::type>(ref);
+  }
+
+  bool empty() {
     return string_value.empty();
   }
 
+  std::type_index getTypeIndex() {
+    return type_id;
+  }
+
+  static const std::type_index UINT64_TYPE;
+  static const std::type_index INT64_TYPE;
+  static const std::type_index INT_TYPE;
+  static const std::type_index BOOL_TYPE;
+  static const std::type_index STRING_TYPE;
+
  protected:
-  std::string string_value;
 
+  template<typename T>
+  bool convertValueImpl(T &ref) {
+    return getValue(ref);
+  }
+
+  template<typename T>
+  void setTypeId() {
+    type_id = std::type_index(typeid(T));
+  }
+
+  virtual bool getValue(int &ref) {
+    ref = std::stol(string_value);
+    return true;
+  }
+
+  virtual bool getValue(int64_t &ref) {
+    ref = std::stoll(string_value);
+    return true;
+  }
+
+  virtual bool getValue(uint64_t &ref) {
+    ref = std::stoull(string_value);
+    return true;
+  }
+
+  virtual bool getValue(bool &ref) {
+    std::istringstream(string_value) >> std::boolalpha >> ref;
```

Comment:
stream >> operator has a return value to detect error, I guess that should be the return value. 

It's also not consistent with the std:stol approach above, which throws.

Reply:
That would be inconsistent with prior behavior I'm afraid.